### ISSUE TITLE
10-line usage example with Eurostat currently broken

### ISSUE THREAD

**Issue by mephinet** [dr-leo/pandaSDMX#237]
Following https://pandasdmx.readthedocs.io/en/latest/example.html does not work any more:
```python
import pandasdmx as sdmx
estat = sdmx.Request('ESTAT')
metadata = estat.datastructure('DSD_une_rt_a')
```
results in:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/gortan/.local/lib/python3.10/site-packages/pandasdmx/api.py", line 473, in get
    response.raise_for_status()
  File "/home/gortan/.local/lib/python3.10/site-packages/requests/models.py", line 1021, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 404 Client Error:  for url: https://ec.europa.eu/eurostat/SDMX/diss-web/rest/datastructure/ESTAT/DSD_une_rt_a/latest?references=all
```

pandasdmx 1.9.0
python 3.10.9

Can you please elaborate whether this is something to be reported to ESTAT-SUPPORT-SDMX@ec.europa.eu or if there was an announced change in the eurostat webservice that needs to be followed up in the pandaSDMX implementation?

**Comment by dr-leo** [comment]
Thanks. This is due to a change of the Eurostat API. It should urgently be reflected I hope to do a new release in the coming days.Am 01.02.2023 um 17:54 schrieb Philipp Gortan ***@***.***>:﻿
Following https://pandasdmx.readthedocs.io/en/latest/example.html does not work any more:
import pandasdmx as sdmx
estat = sdmx.Request('ESTAT')
metadata = estat.datastructure('DSD_une_rt_a')
results in:
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/gortan/.local/lib/python3.10/site-packages/pandasdmx/api.py", line 473, in get
    response.raise_for_status()
  File "/home/gortan/.local/lib/python3.10/site-packages/requests/models.py", line 1021, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 404 Client Error:  for url: https://ec.europa.eu/eurostat/SDMX/diss-web/rest/datastructure/ESTAT/DSD_une_rt_a/latest?references=all

pandasdmx 1.9.0
python 3.10.9
Can you please elaborate whether this is something to be reported to ***@***.*** or if there was an announced change in the eurostat webservice that needs to be followed up in the pandaSDMX implementation?

**Comment by hyperfra** [comment]
Hi

this is the reply by Eurostat

The location of our REST API dissemination endpoint recently changed. The new location is below:
https://ec.europa.eu/eurostat/api/dissemination/sdmx/2.1/

As an SDMX-compliant REST API, the syntax and functioning has remained pretty much unchanged. We support a few more previously unsupported standard SDMX parameters and objects. For the query below, the equivalent query with the new endpoint is
https://ec.europa.eu/eurostat/api/dissemination/sdmx/2.1/dataflow/ESTAT/

**Comment by dr-leo** [comment]
Hi Francesco,

thanks. I hope to release this in a few days. I have also added new
data providers EC_DG_COMP, EMPL and GROW. The end-point for COMP as
shown in ESTAT docs is

https://webgate.ec.europa.eu/comp/redisstat/api/dissemination/sdmx/2.1

But a dataflow query returns an empty message rather than the
catalogue of available dataflows.

Is this because DG COMP and friends have yet to provide data, or am I
mising something?

**Comment by dr-leo** [comment]
I’ve found the issue myself. Thanks anyway.
Leo

Am 23.02.2023 um 22:36 schrieb Leo ***@***.***>:

﻿Hi Francesco,

thanks. I hope to release this in a few days. I have also added new
data providers EC_DG_COMP, EMPL and GROW. The end-point for COMP as
shown in ESTAT docs is

https://webgate.ec.europa.eu/comp/redisstat/api/dissemination/sdmx/2.1

But a dataflow query returns an empty message rather than the
catalogue of available dataflows.

Is this because DG COMP and friends have yet to provide data, or am I
mising something?

**Comment by dr-leo** [comment]
Fixed in v1.10.0 to be released shortly.

**Comment by hyperfra** [comment]
Thanks!!

Sent from my iPhone

On 25 Feb 2023, at 13:27, dr-leo ***@***.***> wrote:

﻿

Fixed in v1.10.0 to be released shortly.
